### PR TITLE
Feature/uar 587 no change scenario json

### PR DESCRIPTION
**Pull request by AndreKramer7654**
### JIRA link

https://companieshouse.atlassian.net/browse/UAR-587

### Change description

Use the noChange statement to produce JSON. The due diligence will be missing and there should be no actual changes.

### Work checklist

- [ ] Tests added where applicable

### Merge instructions

We are committed to keeping commit history clean, consistent and linear. To achieve this, this commit should be structured as follows:

```
<type>[optional scope]: <description>
```

and contain the following structural elements:

- fix: a commit that patches a bug in your codebase (this correlates with PATCH in semantic versioning),
- feat: a commit that introduces a new feature to the codebase (this correlates with MINOR in semantic versioning),
- BREAKING CHANGE: a commit that has a footer `BREAKING CHANGE:` introduces a breaking API change (correlating with MAJOR in semantic versioning). A BREAKING CHANGE can be part of commits of any type,
- types other than `fix:` and `feat:` are allowed, for example `build:`, `chore:`, `ci:`, `docs:`, `style:`, `refactor:`, `perf:`, `test:`, and others,
- footers other than `BREAKING CHANGE: <description>` may be provided.

### CODE REVIEW

File: src/main/java/uk/gov/companieshouse/overseasentitiesapi/model/dto/UpdateDto.java [hunk 1]
```diff
@@ -82,6 +86,10 @@ public void setRegistrableBeneficialOwner(boolean registrableBeneficialOwner) {
         this.registrableBeneficialOwner = registrableBeneficialOwner;
     }
 
+    public boolean isNoChange() { return noChange; }
```
Comment (bantao-ch) [inline src/main/java/uk/gov/companieshouse/overseasentitiesapi/model/dto/UpdateDto.java:89]: probably need to change the formatting ?

File: src/main/java/uk/gov/companieshouse/overseasentitiesapi/validation/OverseasEntitySubmissionDtoValidator.java [hunk 2]
```diff
@@ -78,11 +78,31 @@ private void validateFullUpdateDetails(OverseasEntitySubmissionDto overseasEntit
         // ownersAndOfficersDataBlockValidator.validateOwnersAndOfficers(overseasEntitySubmissionDto, errors, loggingContext);
```
Comment (bantao-ch) [inline src/main/java/uk/gov/companieshouse/overseasentitiesapi/validation/OverseasEntitySubmissionDtoValidator.java:78]: We will need to activate this line as we have now implemented the statements ?

File: src/main/java/uk/gov/companieshouse/overseasentitiesapi/service/FilingsService.java [hunk 3]
```diff
@@ -185,45 +186,72 @@ private void setFilingApiData(
...
     data.put(UPDATE_ENTITY_NUMBER_FIELD, updateSubmission.getEntityNumber());
     data.put(UPDATE_TYPE_FIELD, updateSubmission.getType());
     data.put(UPDATE_USER_SUBMISSION_FIELD, updateSubmission.getUserSubmission());
-    data.put(UPDATE_DUE_DILIGENCE_FIELD, updateSubmission.getDueDiligence());
+    if (!isNoChange) {
+      data.put(UPDATE_DUE_DILIGENCE_FIELD, updateSubmission.getDueDiligence());
+    }
     data.put(UPDATE_PRESENTER_FIELD, updateSubmission.getPresenter());
     data.put(FILING_FOR_DATE_FIELD, updateSubmission.getFilingForDate());
     data.put(NO_CHANGES_IN_FILING_PERIOD_FIELD, updateSubmission.getNoChangesInFilingPeriodStatement());
```
Comment (bbell-companieshouse) [inline src/main/java/uk/gov/companieshouse/overseasentitiesapi/service/FilingsService.java:247]: Does this line need changing at all?

File: src/main/java/uk/gov/companieshouse/overseasentitiesapi/service/FilingsService.java [hunk 4]
```diff
@@ -185,45 +186,72 @@ private void setFilingApiData(
...
+              .addAll(managingOfficerCessationService.managingOfficerCessations(submissionDto,
+                      publicPrivateMoData, logMap));
+      updateSubmission.getAdditions()
+              .addAll(managingOfficerAdditionService.managingOfficerAdditions(submissionDto));
+
+      ApiLogger.debug("Updates have been collected", logMap);
+    }
   }
 
   private void setUpdateSubmissionData(
```
Comment (bbell-companieshouse) [inline src/main/java/uk/gov/companieshouse/overseasentitiesapi/service/FilingsService.java:234]: Not in this ticket scope but Is this method needed for this ticket as the update submission data is populated in the PopulateUpdateSubmission class (called on line 193)

File: src/main/java/uk/gov/companieshouse/overseasentitiesapi/validation/OverseasEntitySubmissionDtoValidator.java (hunk 2 above)
Comment (AndreKramer7654) [reply src/main/java/uk/gov/companieshouse/overseasentitiesapi/validation/OverseasEntitySubmissionDtoValidator.java:78]: separate story for that (uar-711)

File: src/main/java/uk/gov/companieshouse/overseasentitiesapi/service/FilingsService.java (hunk 4 above)
Comment (AndreKramer7654) [reply src/main/java/uk/gov/companieshouse/overseasentitiesapi/service/FilingsService.java:233]: Yes, we don't add the updateSubmission as one but break out all the adds into a HashMap (which is serialized).

File: src/main/java/uk/gov/companieshouse/overseasentitiesapi/service/FilingsService.java (hunk 3 above)
Comment (AndreKramer7654) [reply src/main/java/uk/gov/companieshouse/overseasentitiesapi/service/FilingsService.java]: I've now removed the noChangesInFilingPeriod from the output JSON. The CHIPS filing consumer can just work off the contents - as CHIPS will need to do anyway.

**Review by bbell-companieshouse — APPROVED** [review]

**Review by bantao-ch — APPROVED** [review]

**Pull request merged**